gobblin-hive-registration/build.gradle

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied.
+
+apply plugin: 'java'
+
+dependencies {
+  compile project(":gobblin-api")
+  compile project(":gobblin-utility")
+
+  compile externalDependency.avro
+  compile externalDependency.datanucleusCore
+  compile externalDependency.datanucleusRdbms
+  compile externalDependency.guava
+  compile externalDependency.hiveCommon
+  compile externalDependency.hiveExec
+  compile externalDependency.hiveMetastore
+  compile externalDependency.hiveSerDe
+  compile externalDependency.libthrift
+  compile externalDependency.log4j
+  compile externalDependency.slf4j
+  compile externalDependency.commonsIo
+  compile externalDependency.commonsLang3
+  compile externalDependency.findBugs
+  compile externalDependency.lombok
+  if (project.hasProperty('useHadoop2')) {
+    compile externalDependency.avroMapredH2
+  } else {
+    compile externalDependency.avroMapredH1
+  }
+
+  testCompile externalDependency.calciteCore
+  testCompile externalDependency.calciteAvatica
+  testCompile externalDependency.jhyde
+  testCompile externalDependency.testng
+  testCompile externalDependency.byteman
+  testCompile externalDependency.bytemanBmunit
+  testRuntime externalDependency.derby
+}
+
+task testJar(type: Jar, dependsOn: testClasses) {
+    baseName = "test-${project.archivesBaseName}"
+    from sourceSets.test.output
+}
+
+configurations {
+    tests
+}
+
+configurations {
+    compile {
+        transitive = false
+    }
+}
+
+artifacts {
+    tests testJar
+}
+
+test {
+    useTestNG () {
+      excludeGroups 'ignore'
+      if (project.hasProperty('useHadoop2')) {
+        excludeGroups 'Hadoop1Only'
+      }
+    }
+    workingDir rootProject.rootDir
+    maxParallelForks = 1
+}
+
+clean {
```

Comment:
Is this necessary? Same applies to the `artifacts` and `testJar` closures and the `tests` configuration.